scripts/jest/test-framework-setup.js

```diff
@@ -58,3 +58,26 @@ env.afterEach(() => {
   expect(console.error).toBeReset();
   expect(console.error).toNotHaveBeenCalled();
 });
+
+function wrapDevMatcher(obj, name) {
+  const original = obj[name];
+  obj[name] = function devMatcher() {
+    try {
+      original.apply(this, arguments);
+    } catch (e) {
+      global.__hadDevFailures = e.stack;
+    }
+  };
+}
+
+const expectDev = function expectDev(actual) {
+  const expectation = expect(actual);
+  if (global.__suppressDevFailures) {
```

Comment:
Could this just be `return expectation.not;`?

If we do PROD testing, don't we want to explicitly test that a warning *doesn't* fire?